apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -1007,4 +1009,49 @@ export class ApplicationController {
 
     return { url }
   }
+
+  @Get('applications/:id/attachments/:s3Key/presigned-url')
+  @Scopes(ApplicationScope.read)
+  @Documentation({
+    description: 'Gets a presigned url for attachments',
+    response: { status: 200, type: PresignedUrlResponseDto },
+    request: {
+      query: {},
+      params: {
+        id: {
+          type: 'string',
+          description: 'application id',
+          required: true,
+        },
+        s3Key: {
+          type: 'string',
+          description: 's3 key for attachment',
+          required: true,
+        },
+      },
+    },
+  })
+  async getAttachmentPresignedURL(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @Param('s3Key') s3Key: string,
+    @CurrentUser() user: User,
+  ): Promise<PresignedUrlResponseDto> {
+    const existingApplication = await this.applicationAccessService.findOneByIdAndNationalId(
+      id,
+      user.nationalId,
+    )
+
+    if (!existingApplication.attachments) {
+      throw new NotFoundException('Attachments not found')
+    }
+
+    try {
+      const str = s3Key as keyof typeof existingApplication.attachments
+      const fileName = existingApplication.attachments[str]
+      const { bucket, key } = AmazonS3URI(fileName)
```

Comment:
Nitpick, but this line and maybe even the one above could be inside the `getAttachmentPresignedURL` helper.

Reply:
this one is a bit tricky, maybe this attachments object is not so great to work with ... anyways ... updates are served